src/identity/actions/thunks/index.ts

```diff
@@ -0,0 +1,107 @@
+import {GetState} from 'src/types'
+import {Dispatch} from 'react'
+
+// Actions
+import {
+  setQuartzMe,
+  setQuartzMeStatus,
+  Actions as MeActions,
+} from 'src/me/actions/creators'
+import {
+  setCurrentOrgDetails,
+  setCurrentBillingProvider,
+  setQuartzIdentity,
+  setQuartzIdentityStatus,
+  Actions as IdentityActions,
+} from 'src/identity/actions/creators'
+
+// Types
+import {RemoteDataState} from 'src/types'
+
+// Utilities
+import {
+  fetchQuartzIdentity,
+  fetchAccountDetails,
+  fetchOrgDetails,
+} from 'src/identity/apis/auth'
+import {convertIdentityToMe} from 'src/identity/utils/convertIdentityToMe'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+
+// Thunks
+import {getQuartzMeThunk} from 'src/me/actions/thunks'
+
+export const getQuartzIdentityThunk = () => async dispatch => {
+  if (!isFlagEnabled('quartzIdentity')) {
+    dispatch(getQuartzMeThunk())
+    return
+  }
+
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const quartzIdentity = await fetchQuartzIdentity()
+
+    dispatch(setQuartzIdentity(quartzIdentity))
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Done))
+
+    const legacyMe = convertIdentityToMe(quartzIdentity)
+    dispatch(setQuartzMe(legacyMe, RemoteDataState.Done))
+    dispatch(setQuartzMeStatus(RemoteDataState.Done))
+  } catch (error) {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Error))
+    dispatch(setQuartzMeStatus(RemoteDataState.Error))
+    console.error(error)
+  }
+}
+
+export const getBillingProviderThunk = () => async (
+  dispatch: Dispatch<IdentityActions | MeActions>,
+  getState: GetState
+) => {
+  try {
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Loading))
+
+    const currentState = getState()
+    const accountId = currentState?.identity?.account?.id
+
+    const accountDetails = await fetchAccountDetails(accountId)
+
+    dispatch(setCurrentBillingProvider(accountDetails.billingProvider))
+    dispatch(setQuartzIdentityStatus(RemoteDataState.Done))
+
+    const updatedState = getState()
+    const legacyMe = convertIdentityToMe(updatedState.identity)
+    dispatch(setQuartzMe(legacyMe, RemoteDataState.Done))
+    dispatch(setQuartzMeStatus(RemoteDataState.Done))
```

Comment:
Right now, `convertIdentityToMe` needs to run any time `identity` is updated. 

The selectors in the app all read data from the `quartzMe` state, not the `identity` state (even if the flag is enabled) to minimize risk of breaking anything and amount of code that needs to be changed or guarded by an `if` block. So the flow when the `quartzIdentity` flag is on is always (1) retrieve data from endpoint; (2) update `identity` state; (3) convert our current identity state to the shape of `quartzMe` -- even if we're only changing a couple properties (e.g., the `org` data) or even one property (e.g. `billingProvider`)

Some alternatives:
- Add `if (isFlagEnabled('quartzIdentity'))` throughout the app so we don't need to convert to quartzMe at all when the flag is on.
- Add some new reducers to `quartzMe` so that we can update say, **just** the billingProvider property in the `quartzMe` state, rather than running `convertIdentityToMe` on the whole state any time we fetch a new set of org information or billing provider.

I'd advise against those alternatives - they'd have us adding even more code to support an interface that we're retiring. Running `convertIdentityToMe` isn't computationally expensive, so I think it's okay to have that re-run for now when the identity flag is on.